### ISSUE TITLE
cityscapesLabelTool not open and label images

### ISSUE THREAD

**Issue by EgorAntonovich** [mcordts/cityscapesScripts#50]
Hi, i want to label my images like as cityscape, i use cityscapesLabelTool, when I choose the catalog with images, they don't open...how can i fix it?

i load tool 
![screenshot from 2018-03-26 18-21-45](https://user-images.githubusercontent.com/29814066/37915625-a12bbd06-3122-11e8-92b4-f9cdb4bf3ddb.png)
choose dir
![screenshot from 2018-03-26 18-21-54](https://user-images.githubusercontent.com/29814066/37915654-b4875c84-3122-11e8-94ae-579a979ca95b.png)
and images don't open
![screenshot from 2018-03-26 18-21-59](https://user-images.githubusercontent.com/29814066/37915693-c7c7f560-3122-11e8-91cb-8a20ec445ae6.png)

i can open it with my python script. In it i specify a way to image and tool works.But without label(color)map and i can't label my images.
 open only one picture
![screenshot from 2018-03-26 18-24-44](https://user-images.githubusercontent.com/29814066/37915862-25c8c5ea-3123-11e8-80b7-99be1aaa803a.png)
label area
![screenshot from 2018-03-26 18-24-55](https://user-images.githubusercontent.com/29814066/37915927-3425b76a-3123-11e8-9420-bcab8779c77b.png)
choose class
![screenshot from 2018-03-26 18-25-24](https://user-images.githubusercontent.com/29814066/37915941-434184d6-3123-11e8-9774-d95e3820ff7b.png)
and than don't have results
![screenshot from 2018-03-26 18-25-27](https://user-images.githubusercontent.com/29814066/37915962-506bf75e-3123-11e8-83dd-4f3ce9b1993c.png)
similar situation with bbox
![screenshot from 2018-03-26 18-25-53](https://user-images.githubusercontent.com/29814066/37915980-5c6b573e-3123-11e8-8c91-96496bcad0f1.png)
maybe someone can help me.

**Comment by Bhomik** [comment]
What is the error that you are getting on  the terminal ?

**Comment by EgorAntonovich** [comment]
@Bhomik i don't have any errors when i open with standard script
when i open with my script and try to label one image i have 
Traceback (most recent call last):
  File "/home/egor_antonovich/.local/lib/python2.7/site-packages/cityscapesscripts/annotation/cityscapesLabelTool.py", line 947, in newObject
    self.appendObject( label , self.drawPoly )
  File "/home/egor_antonovich/.local/lib/python2.7/site-packages/cityscapesscripts/annotation/cityscapesLabelTool.py", line 2359, in appendObject
    obj          = CsObject()
TypeError: Can't instantiate abstract class CsObject with abstract methods __str__, fromJsonText, toJsonText

**Comment by Bhomik** [comment]
replace this : 
Line 38 : from annotation import Point, Annotation, CsObject, CsPoly
and replace obj = CsObject() with 
                    obj = CsPoly()

**Comment by EgorAntonovich** [comment]
@Bhomik i replace and now have new errors
Traceback (most recent call last):
  File "/home/egor_antonovich/.local/lib/python2.7/site-packages/cityscapesscripts/annotation/cityscapesLabelTool.py", line 947, in newObject
    self.appendObject( label , self.drawPoly )
  File "/home/egor_antonovich/.local/lib/python2.7/site-packages/cityscapesscripts/annotation/cityscapesLabelTool.py", line 2368, in appendObject
    obj.updateDate()
  File "/home/egor_antonovich/.local/lib/python2.7/site-packages/cityscapesscripts/helpers/annotation.py", line 60, in updateDate
    locale.setlocale( locale.LC_ALL , 'us_us' )
  File "/usr/lib/python2.7/locale.py", line 581, in setlocale
    return _setlocale(category, locale)
locale.Error: unsupported locale setting

**Comment by Bhomik** [comment]
replace in annotation.py with :  
locale.setlocale( locale.LC_ALL , 'en_US.utf8' )

**Comment by EgorAntonovich** [comment]
@Bhomik thanks alot, it work.
why image dir not open?

**Comment by Bhomik** [comment]
it will work, you have to click on icons.

**Comment by Bhomik** [comment]
you can reach me at email : bhomik77@gmail.com if you have further doubts.

**Comment by Shuhui11** [comment]
@Bhomik I am facing similar issue where the tool window opens up but when I clicked on the folder icon, nothing opens up.

<img width="1440" alt="screenshot 2019-02-25 at 9 03 29 am" src="https://user-images.githubusercontent.com/47220371/53308446-59ad4900-38dc-11e9-8879-c4f363df57a8.png">

For your info, I have a CITYSCAPES_DATASET folder on my desktop containing the scripts folder and 1 gtFine and 1 leftImg8bit folder. Has the issue got to do with my folder structure?

Also, I changed a line in the cityscapesLabelTool.py - line 638 from csPath = os.path.join(os.path.dirname(os.path.realpath(**file**)),'..','..') to csPath = os.chdir("/Users/myname/Desktop/CITYSCAPES_DATASET/") as I was having the error message that the data was not found when I clicked on the folder icon initially.

<img width="485" alt="screenshot 2019-02-25 at 9 09 27 am" src="https://user-images.githubusercontent.com/47220371/53308552-39ca5500-38dd-11e9-990b-272fe678ddee.png">

Changing the directory resolves the error message that the data was not found but now nothing pops up. Kindly seek your advice on how do I go about this. Thank you!

**Comment by Shuhui11** [comment]
@Bhomik I am facing similar issue where I draw a polygon, select a class and nothing happens and it prompts me with the same keys again and I can't draw a new polygon.

The code that I am using has been adjusted for based on your previous recommendations to @EgorAntonovich of changing to CsPoly and en_US.utf8 in this issue. @EgorAntonovich have you managed to solve the issue?

I draw a polygon
<img width="1434" alt="screenshot 2019-02-25 at 3 09 29 pm" src="https://user-images.githubusercontent.com/47220371/53320083-77e06c80-390f-11e9-8c87-dccff3b17902.png">

Based on the prompts in the picture above, I typed N and select car and ok
<img width="1439" alt="screenshot 2019-02-25 at 3 09 14 pm" src="https://user-images.githubusercontent.com/47220371/53320098-83339800-390f-11e9-9d19-1e34e1b5e1ab.png">

And it brought me back to the same prompt as the first picture and it is an endless loop.
<img width="1439" alt="screenshot 2019-02-25 at 3 09 14 pm" src="https://user-images.githubusercontent.com/47220371/53320116-95153b00-390f-11e9-86de-56b49826c0cf.png">

For your info, I commented out the Tkinter as of now as it is not compatible with my Mac. Would that affect?
